integrations/tensorflow/bindings/python/pyiree/tf/support/tf_utils.py

```diff
@@ -412,7 +417,10 @@ def _interpret_bytes(tflite_module: bytes, base_dir: str):
     tflite_path = os.path.join(tflite_dir, f"{name}.tflite")
     with open(tflite_path, "wb") as f:
       f.write(tflite_module)
+
     interpreters[name] = tf.lite.Interpreter(tflite_path)
```

Comment:
Does this work? `name` is defined in the loop below and isn't passed to this function. I don't think it will be defined here.

I just tested out an example and it does work apparently... This seems to be reliant on some really weird scoping rules.

```python
>>> def foo():
...   print(name)
... 
>>> def bar():
...   name = "hello"
...   foo()
... 
>>> bar()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 3, in bar
  File "<stdin>", line 2, in foo
NameError: name 'name' is not defined
>>> def foo2():
...   def bar():
...     print(name)
...   name = "hello"
...   bar()
... 
>>> foo2()
hello
```

:vomiting_face:

Let's pass the argument explicitly....

Reply:
This is a pretty common pattern in python (e.g. it was in `compiled_tf_module` with implicit arguments for exactly the same reason before I could flatten the function). I guess the list of arguments we'd need to pass is small enough in this case that it doesn't matter.

Reply:
It added a lot more than I would have expected lol.

```python3
def compile_to_tflite(module_class: Type[tf.Module],
                      exported_names: Sequence[str] = (),
                      artifacts_dir: str = None):
  """Compile a dict of TFLite interpreters for the methods on module_class."""
  functions, names = get_concrete_functions(module_class, exported_names)
  interpreters = dict()
  compiled_paths = None
  if artifacts_dir is not None:
    compiled_paths = dict()

  def _interpret_bytes(tflite_module: bytes, base_dir: str,
                       interpreters: Dict[str, str], compiled_paths: Dict[str,
                                                                          str]):  # oh yapf...
    """Save compiled TFLite module bytes and convert into an interpreter."""
    tflite_dir = os.path.join(base_dir, "tflite")
    os.makedirs(tflite_dir, exist_ok=True)
    tflite_path = os.path.join(tflite_dir, f"{name}.tflite")
    with open(tflite_path, "wb") as f:
      f.write(tflite_module)

    interpreters[name] = tf.lite.Interpreter(tflite_path)
    if compiled_paths is not None:
      compiled_paths[name] = tflite_path
    return interpreters, compiled_paths

  for name, function in zip(names, functions):
    converter = tf.lite.TFLiteConverter.from_concrete_functions([function])
    tflite_module = converter.convert()

    if artifacts_dir is None:
      with tempfile.TemporaryDirectory() as base_dir:
        interpreters, _ = _interpret_bytes(tflite_module, base_dir,
                                           interpreters, None)
    else:
      interpreters, compiled_paths = _interpret_bytes(tflite_module,
                                                      artifacts_dir,
                                                      interpreters,
                                                      compiled_paths)

  return interpreters, compiled_paths
```